lib/LANraragi/Plugin/Hdoujin.pm

```diff
@@ -0,0 +1,192 @@
+    shift;
+    my ( $title, $tags, $thumbhash, $file, $oneshotarg, @args ) = @_;
+
+    my $logger = LANraragi::Utils::Generic::get_logger( "Hdoujin", "plugins" );
+
+    if ( LANraragi::Utils::Archive::is_file_in_archive( $file, "info.json" ) ) {
+
+        #Extract info.json
+        my $filepath = LANraragi::Utils::Archive::extract_file_from_archive( $file,
+            "info.json" );
+
+        #Open it
+        my $stringjson = "";
+
+        open( my $fh, '<:encoding(UTF-8)', $filepath )
+          or return ( error => "Could not open $filepath!" );
+
+        while ( my $row = <$fh> ) {
+            chomp $row;
+            $stringjson .= $row;
+        }
+
+        #Use Mojo::JSON to decode the string into a hash
+        my $hashjson = from_json $stringjson;
+
+        $logger->debug("Found and loaded the following JSON: $stringjson");
+
+        #Parse it
+        $tags = tags_from_Hdoujin_json($hashjson);
+
+        #Delete it
+        unlink $filepath;
+
+        #Return tags
+        $logger->info("Sending the following tags to LRR: $tags");
+        return ( tags => $tags );
+
+    }
+    else {
+
+        return ( error => "No Hdoujin info.json file found in this archive!" );
+    }
+
+}
+
+#tags_from_Hdoujin_json(decodedjson)
+#Goes through the JSON hash obtained from an info.json file and return the contained tags.
+sub tags_from_Hdoujin_json {
+
+    	my $hash = $_[0];
+	my $hashcontrol = $_[0];
+    	my $return = "";
+	my $tagwRespect = $_[0];
+	my $tagnoRespect = $_[0];
+	my $choosehash = "0";
+	
+	
+	my $tagscontrol = $hashcontrol->{"manga_info"}->{"tags"};
+	
+		if (ref($tagscontrol) eq 'HASH'){
+		$choosehash = "1";
+		}
+	
+#HDoujin jsons are composed of a main manga_info object, containing fields for every metadata.
+#Those fields can contain either a single tag or an array of tags.
+	
+    	my $tags = $hash->{"manga_info"};
+
+    	#Take every key in the manga_info hash, except for title which we're already processing
+	
+	my @filtered_keys = grep { $_ ne "tags" and $_ ne "title" } keys(%$tags);
+
+	foreach my $namespace ( @filtered_keys ) {
+
+        my $members = $tags->{$namespace};
+	   
+	    if(ref($members) eq 'ARRAY'){
+    
+
+			foreach my $tag (@$members){
+
+				$return .= ", " unless $return eq "";
+				$return .= $namespace . ":" . $tag unless $members eq "";
+			
+			} 
+			
+		}
+		else {
+		
+				$return .= ", " unless $return eq "";
+				$return .= $namespace . ":" . $members unless $members eq "";
+				
+		}
+	
+	}
+    
+		if( $choosehash == "1"){
+	
+				return $return . "," . tags_from_wRespect($tagwRespect);
+		
+		}
+		else {
+		
+				return $return . "," . tags_from_noRespect($tagnoRespect);
+				
+		}
+
+}
+
+sub tags_from_wRespect {
+
+    my $hash   = $_[0];
+    my $return = "";
+    my $tags = $hash->{"manga_info"}->{"tags"};
+
+    foreach my $namespace ( keys(%$tags) ) {
+
+        my $members = $tags->{$namespace};
+        foreach my $tag (@$members) {
+
+	    $namespace = "Tags";

```

Comment:
Unnecessary namespace imo: It's cleaner at this point to just add the raw tags.

Reply:
I have added this because, example with nhentai every tags goes on the "misc" field (misc:something) and it was visually not good imo, and so with the namespace tags every name become tags:something, like the json file without the option respect on.
now, I thinked and tried with e-hentai, that save the tag like female:something and male:something, so with that namespace renaming I will lost that information, so I think that line is an error.

Reply:
I personally prefer having unnamespaced tags when there's no precise namespace rather than adding a generic one, but it doesn't really matter I guess.